lib/entity_store/controls/entity_store.rb

```diff
@@ -47,7 +47,7 @@ def self.example_class(category: nil, entity_class: nil, projection_class: nil,
           include ::EntityStore
 
           category category
-          entity entity_class
+          entity entity_class, id_attribute: entity_id_attribute
```

Comment:
I opted for just allowing setting the ID rather than specifying a block. A block would be more flexible and allow for instantiating hashes, or arrays, or whatever you want. 

Originally, I considered allowing a block or a class to be specified, but there was logging tied to the actual class that seemed important.

Another consideration is the name of the property. I included the word `attribute`, but it could very well be superfluous: `entity entity_class, id: :some_id` and `store.entity_id == :some_id`